### PR TITLE
[EngSys] added to python script to not fail if there is an existing PR

### PR DESCRIPTION
**Pull request by seankane-msft**
This PR adds to the `verify_autorest.py` file to only fail if the autorest is not the same AND there is not an existing PR out for autorest generation.

### CODE REVIEW

**Comment by seankane-msft** [comment]
#18393

File: scripts/devops_tasks/verify_autorest.py [hunk 1]
```diff
@@ -54,7 +55,31 @@ def find_swagger_folders(directory):
     return ret
 
 
-def check_diff(folder):
+def check_for_open_pr(service_directory):
```
Comment (weshaggard) [inline scripts/devops_tasks/verify_autorest.py:58]: You shouldn't need to do this if your branch name is consistent. As the create-pull-request template will not create an additional PR against a given branch if one already exists. That is how we generally handle this case in other places.

File: scripts/devops_tasks/verify_autorest.py [hunk 2]
```diff
@@ -54,7 +55,31 @@ def find_swagger_folders(directory):
...
+        for link in body:
+            if link["title"] == pr_title:
+                return True
+
+        next_page = response.links["next"]["url"]
+        response = s.get(next_page)
+    return False
+
+
+def check_diff(folder, service_dir):
```
Comment (weshaggard) [inline scripts/devops_tasks/verify_autorest.py:82]: Similar you shouldn't need to manually do a git diff.  See https://github.com/Azure/azure-sdk-for-python/blob/master/eng/common/pipelines/templates/steps/git-push-changes.yml#L18

File: scripts/devops_tasks/verify_autorest.py (hunk 1 above)
Comment (seankane-msft) [reply scripts/devops_tasks/verify_autorest.py:58]: That is great to know, I can close this PR entirely then

File: scripts/devops_tasks/verify_autorest.py (hunk 2 above)
Comment (seankane-msft) [reply scripts/devops_tasks/verify_autorest.py:82]: The git diff is for users to be able to inspect which files are different in the pipeline on a failure.

Comment (weshaggard) [reply scripts/devops_tasks/verify_autorest.py:82]: OK if that is interesting for other cases that is fine. However I'd encourage you to do those type of command line calls inline script blocks in the yml file, unless you feel folks would want to run them locally as well. 

This is perhaps my preference for using a shell scripting language (script/bash/pwsh) when running command line tools as opposed to wrapping them in process calls in a higher level language like python. Just something to keep in mind not something that is necessarily actionable. :)

**Pull request closed without merging**